vllm/engine/llm_engine.py

```diff
@@ -1442,6 +1551,11 @@ def _get_stats(
 
             for idx, scheduled_seq_group in enumerate(
                     scheduler_outputs.scheduled_seq_groups):
+                # Skip double logging when using async output proc
+                if finished_before and idx in finished_before:
+                    actual_num_batched_tokens -= 1
+                    continue
```

Comment:
I can't connect the comment and the code. Can you elaborate it?

Reply:
`actual_num_batched_tokens -= 1` is required to avoid double-counting. With async output proc, number of batched tokens is already counted since its lagged by a step. If we have a token that was finished before, we won't count it again. This does not happen with non-async logging.